.github/workflows/codeql.yml

```diff
@@ -3,10 +3,7 @@ name: "CodeQL"
 on:
   push:
     branches: [ "master", "develop", "release-*" ]
-  pull_request:
-    branches: [ "develop" ]
-
-permissions: read-all
```

Comment:
Need to remove this because the only job in here anyway sets it to fairly limited. However in the top level we need to set the subjob permissions to allow that one "write" permission. That overwrites the default settings for everything else (read-all) and you have to set every sub thing manually or it will default to "none". However you can't set id-token to "read" that will result in a failed run. But you also can't set it to "none" in the parent, if the child sets "read-all" because "read-all" does in fact set id-token to "read" even though one is not allowed to set it explicitly to "read" ...